### ISSUE TITLE
Grid learning with gradient descent

### ISSUE THREAD

**Issue by Indoxer** [KindXiaoming/pykan#102]
My question is about this line in the paper:

"Other possibilities are: (a) the grid is learnable with gradient descent, e.g., [16];" 

There is an implementation of this?

Also, I noticed that torch.linalg.lstsq can only work with full rank matrices on cuda. 
Does anyone know an implementation of this in cuda?

PS I'm working on the "library", where I'm testing various ideas that I found in issues, e.g. training on MNIST, CIFAR10, KAN as convolution (in progress), etc. (https://github.com/Indoxer/LKAN). I am open to contribution.

**Comment by KindXiaoming** [comment]
close for now, feel free to open if there's further update
